Packs/AzureKeyVault/Integrations/AzureKeyVault/AzureKeyVault.py

```diff
@@ -0,0 +1,1304 @@
+
+        if ip_rules:
+            network_acls["ipRules"] = []
+            for ip in ip_rules:
+                network_acls["ipRules"].append({'value': ip})
+
+        return network_acls
+
+    def config_vault_properties(self, object_id: str, tenant_id: str, enabled_for_deployment: bool,
+                                enabled_for_disk_encryption: bool,
+                                enabled_for_template_deployment: bool, sku_name: str,
+                                permissions: Dict[str, Any], network_acls: Dict[str, Any]):
+
+        """
+        Configures the properties of a vault on create or update command.
+
+        Args:
+            object_id (str): The object ID of a user, service principal or security group in the Azure Active Directory.
+            tenant_id (str):An identity that have access to the key vault.
+            enabled_for_deployment (bool): permission for Azure VM to retrieve certificates stored as secrets.
+            enabled_for_disk_encryption (bool): permission for Azure Disk Encryption to retrieve secrets.
+            enabled_for_template_deployment (bool): permission for Azure Resource Manager to retrieve secrets.
+            sku_name (str):Sku name.
+            permissions (Dict[str,Any]): Key Vault access policy property.
+            network_acls (Dict[str,Any]): Key Vault network acls property.
+
+        Returns:
+            Dict[str,Any]: Key Vault properties.
+        """
+        properties = {"accessPolicies": [
+            {"objectId": object_id, "permissions": permissions,
+             "tenantId": tenant_id}],
+            "enabledForDeployment": enabled_for_deployment,
+            "enabledForDiskEncryption": enabled_for_disk_encryption,
+            "enabledForTemplateDeployment": enabled_for_template_deployment,
+            "sku": {"family": "A", "name": sku_name}, "tenantId": tenant_id}
+
+        if network_acls:
+            properties["networkAcls"] = network_acls
+
+        return properties
+
+    def get_entities_independent_of_pages(self, first_page: Dict[str, Any], limit: int, offset: int,
+                                          resource: str = MANAGEMENT_RESOURCE) -> List[dict]:
+        """
+        List the entities independent of azure's page size.
+
+        Args:
+            first_page (Dict[str, Any]): The first list of entities which returned by the first API call.
+            limit (int): limit on the number of entities to retrieve to the user.
+            offset (int): first index to return from.
+            resource (str): Azure resource. Default is MANAGEMENT_RESOURCE.
+
+        Returns:
+            List[Dict,Any]: List of Key Vaults/Keys/Secrets/Certificates.
+        """
+        entities = first_page.get('value')
+        next_page_url = first_page.get('nextLink')
+        # more entities to get
+        while next_page_url and len(entities) < offset + limit:
```

Comment:
Since the entities variable holds all the entities which retrieved from the first API call (and not starting from the offset index) the while statement must take it into consideration.